src/main/java/org/elasticsearch/index/query/SimpleQueryStringParser.java

```diff
@@ -221,6 +226,10 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         if (queryName != null) {
             parseContext.addNamedQuery(queryName, query);
         }
+
+        if (minimumShouldMatch != null && query instanceof BooleanQuery) {
```

Comment:
I don't think this is the best place to apply minShouldMatch. The difference between this and say match query is that it supports multiple fields (i realize the default is _all and this patch will work with that).

So in this case, this query will return a boolean query across say title, body, author fields. the current application of minShouldMatch will be unintuitive there.